python_modules/libraries/dagster-aws/dagster_aws/ecs/executor.py

```diff
@@ -0,0 +1,394 @@
+        step_key = self._get_step_key(step_handler_context)
+
+        if step_handler_context.execute_step_args.known_state:
+            retry_count = step_handler_context.execute_step_args.known_state.get_retry_state().get_attempt_count(
+                step_key
+            )
+        else:
+            retry_count = 0
+
+        return "%s-%d" % (step_key, retry_count)
+
+    def _get_step_key(self, step_handler_context: StepHandlerContext) -> str:
+        step_keys_to_execute = cast(
+            List[str], step_handler_context.execute_step_args.step_keys_to_execute
+        )
+        assert len(step_keys_to_execute) == 1, "Launching multiple steps is not currently supported"
+        return step_keys_to_execute[0]
+
+    def _get_container_context(
+        self, step_handler_context: StepHandlerContext
+    ) -> EcsContainerContext:
+        return EcsContainerContext.create_for_run(
+            step_handler_context.dagster_run,
+            cast(EcsRunLauncher, step_handler_context.instance.run_launcher),
+        )
+
+    def _run_task(self, **run_task_kwargs):
+        return run_ecs_task(self.ecs, run_task_kwargs)
+
+    def launch_step(self, step_handler_context: StepHandlerContext) -> Iterator[DagsterEvent]:
+        step_key = self._get_step_key(step_handler_context)
+
+        step_tags = step_handler_context.step_tags[step_key]
+
+        container_context = self._get_container_context(step_handler_context)
+
+        run = step_handler_context.dagster_run
+
+        args = step_handler_context.execute_step_args.get_command_args(
+            skip_serialized_namedtuple=True
+        )
+
+        run_task_kwargs = self._get_run_task_kwargs(
+            run,
+            args,
+            step_key,
+            step_tags,
+            step_handler_context=step_handler_context,
+            container_context=container_context,
+        )
+
+        task = backoff(
+            self._run_task,
+            retry_on=(RetryableEcsException,),
+            kwargs=run_task_kwargs,
+            max_retries=int(
+                os.getenv("STEP_TASK_RETRIES", DEFAULT_STEP_TASK_RETRIES),
+            ),
+        )
+
+        yield DagsterEvent.step_worker_starting(
+            step_handler_context.get_step_context(step_key),
+            message=f'Executing step "{step_key}" in ECS task.',
+            metadata={
+                "Task ARN": MetadataValue.text(task["taskArn"]),
+            },
+        )
+
+        step_id = self._get_step_id(step_handler_context)
+
+        self._launched_tasks[step_id] = task["taskArn"]
+
+    def check_step_health(self, step_handler_context: StepHandlerContext) -> CheckStepHealthResult:
+        step_key = self._get_step_key(step_handler_context)
+
+        task_arn = self._launched_tasks.get(self._get_step_id(step_handler_context))
+        cluster_arn = self._cluster_arn
```

Comment:
probably want some more explicit error handling here if a task with that ID can't be found ( should never happen )

Reply:
done